src/engine/cachingreader/cachingreaderworker.cpp

```diff
@@ -93,15 +91,21 @@ void CachingReaderWorker::run() {
     while (!atomicLoadAcquire(m_stop)) {
         // Request is initialized by reading from FIFO
         CachingReaderChunkReadRequest request;
-        if (m_newTrackAvailable) {
+        if (atomicLoadAcquire(m_newTrackAvailable)) {
```

Comment:
m_newTrackAvailable.loadAcquire()? The compatibility functions doesn't seem to be needed anymore.